test/e2e/messaging.test.ts

```diff
@@ -0,0 +1,170 @@
+        .expect(204)
+
+      // poll until a completed connection appears
+      await pollGetConnections(
+        config.alice.veritableUrl,
+        (connections) =>
+          connections.length === 1 && connections[0].state === 'completed'
+      )
+
+      const { body: aliceConnections } = await request(
+        config.alice.veritableUrl
+      ).get('/connections')
+
+      const aliceConnectionRecord = aliceConnections[0] as ConnectionRecord
+      expect(aliceConnectionRecord.invitationDid).to.equal(config.bob.did)
+
+      // poll until a completed connection appears
+      await pollGetConnections(
+        config.bob.veritableUrl,
+        (connections) =>
+          connections.length === 1 && connections[0].state === 'completed'
+      )
+
+      const { body: bobConnections } = await request(
+        config.bob.veritableUrl
+      ).get(`/connections`)
+      const bobConnectionRecord = bobConnections[0] as ConnectionRecord
+      expect(bobConnectionRecord.state).to.equal('completed')
+      expect(bobConnectionRecord.did).to.equal(aliceConnectionRecord.theirDid)
+
+      aliceTestDid = bobConnectionRecord.theirDid
+      bobTestDid = bobConnectionRecord.did
+
+      //send message to Bob from alice
+      await aliceClient
+        .post('/messages')
+        .send({
+          message: {
+            content: 'Alice says hi!',
+          },
+          did: aliceConnectionRecord.theirDid,
+        })
+        .expect(204)
+    })
+    it('Alice gets messages from her side to check if it has been sent.', async function () {
+      const res = await aliceClient.get(`/messages`)
+
+      expect(res.body).to.be.an('array').and.to.have.lengthOf(1)
+      expect(res.body).to.deep.include({
+        content: 'Alice says hi!',
+        recipientDid: bobTestDid,
+      })
+    })
+    it('Bob gets messages from his side to check if it has been received.', async function () {
+      const res = await bobClient.get(`/messages`)
+      expect(res.body).to.be.an('array').and.to.have.lengthOf(1)
+      expect(res.body).to.deep.include({
+        content: 'Alice says hi!',
+        senderDid: aliceTestDid,
+      })
+    })
+    it('Bob responds to message from Alice.', async function () {
+      await bobClient
+        .post('/messages')
+        .send({
+          message: {
+            content: 'Bob says hi back!',
+          },
+          did: aliceTestDid,
+        })
+        .expect(204)
+    })
+    it('Alice gets messages from her side to check if she now also has response.', async function () {
+      const res = await aliceClient.get(`/messages`)
+
+      expect(res.body).to.be.an('array').and.to.have.lengthOf(2)
+      expect(res.body).to.deep.include({
+        content: 'Alice says hi!',
+        recipientDid: bobTestDid,
+      })
+      expect(res.body).to.deep.include({
+        content: 'Bob says hi back!',
+        senderDid: bobTestDid,
+      })
+    })
+  })
+})
+
+const pollGetConnections = async (
```

Comment:
can reuse the existing function on `main`, maybe it needs moving out into some shared/utils directory